### PR TITLE
Fix for DefaultSamCliProcessInvoker not treating string input properly

### PR DESCRIPTION
**Pull request by awschristou**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)

## Description

Intended as a temporary fix that doesn't prevent sam cli commands from running

Also: Touched up sam build and sam info logging.

## Testing

* Ran command that validates sam version
* Ran local run and local debug from codeLens
* Ran command that creates new sam application

## Checklist

- [x] I have read the **README** document
- [x] I have read the **CONTRIBUTING** document
- [x] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed
- [ ] A short description of the change has been added to the **CHANGELOG**

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/shared/sam/cli/samCliInfo.ts [hunk 1]
```diff
@@ -27,22 +27,20 @@ export class SamCliInfoInvocation {
...
 
-        let errorMessage: string | undefined
-        if (!!childProcessResult.error && !!childProcessResult.error.message) {
-            errorMessage = childProcessResult.error.message
-        } else if (!!childProcessResult.stderr) {
-            errorMessage = childProcessResult.stderr
-        }
-
-        throw new Error(`sam --info encountered an error: ${errorMessage}`)
+        throw new Error(`sam --info encountered an error: ${
```
Comment (mpiroc) [inline src/shared/sam/cli/samCliInfo.ts]: `childProcessResult.error && childProcessResult.error.message` evaluates to a `boolean`, so if neither are falsy, the message will be:

    sam --info encountered an error: true

File: src/shared/sam/cli/samCliInvoker.ts [hunk 2]
```diff
@@ -35,6 +35,8 @@ export class DefaultSamCliProcessInvoker implements SamCliProcessInvoker {
         let options: SpawnOptions | undefined
         if (Array.isArray(optionsOrArgs)) {
             args = optionsOrArgs as string[]
+        } else if (typeof optionsOrArgs === 'string' || optionsOrArgs instanceof String) {
```
Comment (mpiroc) [inline src/shared/sam/cli/samCliInvoker.ts]: This doesn't seem correct...

1. `optionsOrArgs` is of type `SpawnOptions` or `string[]`, never `string`.
2. `instanceof` is redundant, and will cause an error if `optionsOrArgs` is undefined.
3. If `optionsOrArgs` *could* be a string, the compiler is aware of `typeof` causes for build-in types, so you wouldn't need to call `optionsOrArgs.toString()`

I'll put together a PR with the fix.

File: src/shared/sam/cli/samCliBuild.ts [hunk 3]
```diff
@@ -35,9 +35,10 @@ export class SamCliBuildInvocation {
         console.error('SAM CLI error')
         console.error(`Exit code: ${exitCode}`)
         console.error(`Error: ${error}`)
+        console.error(`stderr: ${stderr}`)
         console.error(`stdout: ${stdout}`)
 
-        throw new Error(`sam build encountered an error: ${error && error.message ? error.message : stdout}`)
+        throw new Error(`sam build encountered an error: ${error && error.message || stderr || stdout}`)
```
Comment (mpiroc) [inline src/shared/sam/cli/samCliBuild.ts]: `error && error.message` evaluates to a `boolean`, so if neither are falsey, the message will be:

    sam build encountered an error: true

**Comment by awschristou** [comment]
Closing in favor of #233

**Pull request closed without merging**